NETOBSERV-915 add conversation direction

jpinsonneau opened the pull request:
This PR adds direction fields to conversation tracking

Also, it adds JSON fields sorting to be able to easy compare 2 items

Related PRs:
- https://github.com/netobserv/network-observability-operator/pull/313
- https://github.com/netobserv/flowlogs-pipeline/pull/430

jpinsonneau commented:
/hold

jpinsonneau commented:
/hold

jotak commented on web/src/components/netflow-record/record-panel.tsx:
```diff
@@ -225,6 +225,12 @@ export const RecordPanel: React.FC<RecordDrawerProps> = ({
 
+  const getSortedJSON = React.useCallback(() => {
+    const allKeys = new Set<string>();
+    JSON.stringify(record, (key, value) => (allKeys.add(key), value));
```
This `stringify` call seems useless? (return value is ignored)
Could it be simplified with just: 
```
return JSON.stringify(record, Object.keys(record).sort(), 2); 
```
?

jpinsonneau replied:
that's a trick to list all keys of `record`
https://stackoverflow.com/a/53593328

else you would need to implement an [iterator to get subkeys](https://stackoverflow.com/a/16168003) (record.labels & record.fields)

jotak replied:
This then: ?

```
  const flat = { ...record.fields, ...record.labels };
  return JSON.stringify(flat, Object.keys(flat).sort(), 2);
```

(sorry but the double-stringify looks really hacky :-) )

jpinsonneau replied:
done

Amoghrd commented:
/ok-to-test

jpinsonneau commented:
Addressed feedback on json sort + include `_RecordType` as secondary criteria to sort start / end time columns to ensure start < tick < end order when logs have exact same time:

![image](https://user-images.githubusercontent.com/91894519/235897067-026fcd24-d0fb-4727-bb92-cef7b3c1ba52.png)

Amoghrd commented:
/ok-to-test

Amoghrd commented:
/ok-to-test

Amoghrd commented:
/label qe-approved

jpinsonneau commented:
/approve
